v03_pipeline/lib/tasks/update_variant_annotations_table_with_new_samples.py

```diff
@@ -161,92 +84,47 @@ def complete(self) -> bool:
-            self.reference_genome,
-        )
-
-        # 2) Select down to the formatting annotations fields and
-        # any reference dataset collection annotations.
-        new_variants_ht = new_variants_ht.select(
-            **get_fields(
-                new_variants_ht,
-                self.dataset_type.formatting_annotation_fns(self.reference_genome),
-                **self.rdc_annotation_dependencies,
-                **self.other_annotation_dependencies,
-                **self.param_kwargs,
+        new_variants_ht = hl.read_table(
+            new_variants_table_path(
+                self.reference_genome,
+                self.dataset_type,
+                self.run_id,
             ),
         )
 
-        # 3) Join against the reference dataset collections that are not "annotated".
-        for rdc in ReferenceDatasetCollection.for_reference_genome_dataset_type(
-            self.reference_genome,
-            self.dataset_type,
-        ):
-            if rdc.requires_annotation:
-                continue
-            rdc_ht = self.rdc_annotation_dependencies[f'{rdc.value}_ht']
-            new_variants_ht = new_variants_ht.join(rdc_ht, 'left')
-
-        # 4) Union with the existing variant annotations table
-        # and annotate with the lookup table.
+        # Union with the new variants table and annotate with the lookup table.
         ht = ht.union(new_variants_ht, unify=True)
         if self.dataset_type.has_lookup_table:
-            ht = ht.annotate(
+            callset_ht = get_callset_ht(
+                self.reference_genome,
+                self.dataset_type,
+                self.callset_paths,
+                self.project_guids,
+                self.project_remap_paths,
+                self.project_pedigree_paths,
+            )
+            # new_variants_ht consists of variants present in the new callset, fully annotated,
+            # but NOT present in the existing annotations table.
+            # callset_variants_ht consists of variants present in the new callset, fully annotated,
+            # and either present or not present in the existing annotations table.
+            callset_variants_ht = ht.semi_join(callset_ht)
+            ht = ht.anti_join(callset_ht)
+            callset_variants_ht = callset_variants_ht.annotate(
                 **get_fields(
-                    ht,
+                    callset_variants_ht,
                     self.dataset_type.lookup_table_annotation_fns,
-                    **self.rdc_annotation_dependencies,
-                    **self.other_annotation_dependencies,
+                    lookup_ht=hl.read_table(
+                        lookup_table_path(
+                            self.reference_genome,
+                            self.dataset_type,
+                        ),
+                    ),
                     **self.param_kwargs,
                 ),
             )
+            ht = ht.union(callset_variants_ht, unify=True)
```

Comment:
I had to add `unify=True` to handle the theoretical case if `ht` is empty and all the variants are new, `gt_stats` won't be in a column in the table yet and `union` fails but I think `unify=True` makes sense for this case.